core/templates/components/skill-mastery/skill-mastery.component.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the skill mastery viewer.
+ */
+
+import { SkillMasteryListConstants } from
+  'components/skills-mastery-list/skills-mastery-list.constants';
+
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { SkillMasteryBackendApiService } from
+  'domain/skill/skill-mastery-backend-api.service';
+
+@Component({
+  selector: 'skill-mastery-viewer',
+  templateUrl: './skill-mastery.directive.html',
+  styleUrls: []
+})
+export class SkillMasteryViewerComponent implements OnInit {
+  @Input() skillId: string;
+  @Input() masteryChange: number;
+
+  skillMasteryDegree: number = 0;
+
+  constructor(
+    private skillMasteryBackendApiService: SkillMasteryBackendApiService
+  ) {}
+
+  ngOnInit() {
+    this.skillMasteryDegree = 0.0;
+
+    this.skillMasteryBackendApiService.fetchSkillMasteryDegrees(
+      [this.skillId]).then(function(degreesOfMastery) {
+      this.skillMasteryDegree = degreesOfMastery[this.skillId];
+    });
+  }
+
+  getSkillMasteryPercentage() {
+    return Math.round(this.skillMasteryDegree * 100);
+  }
+
+  getMasteryChangePercentage() {
+    if (this.masteryChange >= 0) {
+      return '+' + Math.round(this.masteryChange * 100);
+    } else {
+      return Math.round(this.masteryChange * 100);
+    }
+  }
+
+  getLearningTips() {
+    if (this.masteryChange > 0) {
+      if (this.skillMasteryDegree >=
+          SkillMasteryListConstants.MASTERY_CUTOFF.GOOD_CUTOFF) {
+        return 'You have mastered this skill very well! ' +
+          'You can work on other skills or learn new skills.';
```

Comment:
Wrap multi-line strings in parentheses.
```suggestion
        return ('You have mastered this skill very well! ' +
          'You can work on other skills or learn new skills.');
```

Reply:
Done